specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-11-15-preview/restorable.json

```diff
@@ -0,0 +1,1863 @@
+            "name": "location",
+            "required": true,
+            "description": "The name of the continuous backup restore location.",
+            "schema": {
+              "$ref": "#/definitions/ContinuousBackupRestoreLocation"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Table retrieve backup information operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/BackupInformation"
+            }
+          },
+          "202": {
+            "description": "The Table retrieve backup information operation will complete asynchronously."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "cosmos-db.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/locations/{location}/restorableDatabaseAccounts/{instanceId}/restorableTables": {
+      "get": {
+        "operationId": "RestorableTables_List",
+        "x-ms-examples": {
+          "CosmosDBRestorableTableList": {
+            "$ref": "./examples/CosmosDBRestorableTableList.json"
+          }
+        },
+        "description": "Show the event feed of all mutations done on all the Azure Cosmos DB Tables. This helps in scenario where table was accidentally deleted. This API requires 'Microsoft.DocumentDB/locations/restorableDatabaseAccounts/.../read' permission",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/locationParameter"
+          },
+          {
+            "$ref": "#/parameters/instanceIdParameter"
+          },
+          {
+            "$ref": "#/parameters/restorableTablesEventFeedStartTime"
+          },
+          {
+            "$ref": "#/parameters/restorableTablesEventFeedEndTime"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/RestorableTablesListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "cosmos-db.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/locations/{location}/restorableDatabaseAccounts/{instanceId}/restorableTableResources": {
```

Comment:
This should be updated to follow the same pattern as restorableMongodbResources.  Its missing x-ms-pageable and response schema is not following the appropriate GET collection schema `{ "value": [] }`

Reply:
response for this API is bit different than MongoDB api. We return the response in following format.
{
TableNames: []
}
That's why it doesn't have x-ms-pageable header.

Reply:
graphResources also follows the correct pattern.  The schema you describe is not the correct ARM schema for a collection GET like this (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource)

Reply:
Sql, Mongo and Graph api contain both database and collections so response looks as below:
[
  {
     "DatabaseName:"db1",
     "CollectionNames": ["col1",...]
  },
  ...
]

But, for tables we don't have databasename, so we just return the table names:
TableNames: ["table1",...]

Reply:
I can also find some other apis in cosmos-db.json which are get and not pageable. for ex:

  "paths": {
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}": {
      "get": {
              ....
        ],
        "responses": {
          "200": {
            "description": "The database account properties were retrieved successfully.",
            "schema": {
              "$ref": "#/definitions/DatabaseAccountGetResults"
            }
          }
        }
      },

Reply:
Chris, we already have management changes for it and we want to release the preview in this release for preview. Is it okay, we can get these changes in preview and then i make the suggested changes in stable api version?

Reply:
Okay, Chris.
I'll make that change. Could you please also point out any other changes required so that i can make the management changes for those in one PR and still try to meet the release train.

Reply:
(I made the same changes first in[ stable api version](https://github.com/Azure/azure-rest-api-specs/pull/17380) and got approval like 20 days ago. But later, we decided that we are only going to make preview changes so didn't merge that PR. Based on that approval, I was under the assumption that it shouldn't be any issue in this PR as well)

Reply:
Hi Chris, I have updated the PR with the suggested changes. Please review the rest of the changes if they looks okay so that i can combine the service side changes for that in one PR.